File: extensions/shared/library/src/main/java/app/morphe/extension/shared/patches/DrcAudioPatch.java
package app.morphe.extension.shared.patches;

import app.morphe.extension.shared.settings.BaseSettings;

@SuppressWarnings("unused")
public final class DrcAudioPatch {

    /**
     * Checks if DRC audio should be disabled according to user settings.
     */
    public static boolean disableDrcAudio() {
        return BaseSettings.DISABLE_DRC_AUDIO.get();
    }

    /**
     * Override volume normalization feature flag.
     */
    public static boolean disableDrcAudioFeatureFlag(boolean original) {
        return !BaseSettings.DISABLE_DRC_AUDIO.get() && original;
    }
}

File: patches/src/main/kotlin/app/morphe/patches/youtube/video/audio/DrcAudioPatch.kt
package app.morphe.patches.youtube.video.audio

import app.morphe.patches.shared.misc.drc.drcAudioPatch
import app.morphe.patches.youtube.misc.extension.sharedExtensionPatch
import app.morphe.patches.youtube.misc.playservice.versionCheckPatch
import app.morphe.patches.youtube.misc.settings.PreferenceScreen
import app.morphe.patches.youtube.misc.settings.settingsPatch

@Suppress("unused")
val drcAudioPatch = drcAudioPatch(
    block = {
        dependsOn(
            sharedExtensionPatch,
            settingsPatch,
            versionCheckPatch
        )

        compatibleWith(
            "com.google.android.youtube"(
                "20.14.43",
                "20.21.37",
                "20.26.46",
                "20.31.42",
                "20.37.48",
                "20.40.45",
            )
        )
    },
    preferenceScreen = PreferenceScreen.VIDEO,
)
